### PR TITLE
[Merged by Bors] - feat(SetTheory/Cardinal/Aleph): `range aleph = Ici ℵ₀`

### PR DESCRIPTION
**Pull request by vihdzp**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/SetTheory/Cardinal/Aleph.lean [hunk 1]
```diff
@@ -334,6 +334,9 @@ theorem preAleph_omega0 : preAleph ω = ℵ₀ :=
 theorem aleph0_le_preAleph {o : Ordinal} : ℵ₀ ≤ preAleph o ↔ ω ≤ o := by
   rw [← preAleph_omega0, preAleph_le_preAleph]
 
+theorem range_preAleph : range preAleph = Set.univ :=
```
Comment (b-mehta) [inline Mathlib/SetTheory/Cardinal/Aleph.lean:337]: This one doesn't seem that necessary to me - `simp` already knows it, and `rw?` gets it too, the specialisation here appears unhelpful.

File: Mathlib/SetTheory/Cardinal/Aleph.lean [hunk 2]
```diff
@@ -431,7 +447,9 @@ theorem preAleph_isNormal : IsNormal (ord ∘ preAleph) :=
 theorem aleph_isNormal : IsNormal (ord ∘ aleph) :=
   preAleph_isNormal.trans <| isNormal_add_right ω
 
-theorem succ_aleph0 : succ ℵ₀ = ℵ₁ := by rw [← aleph_zero, ← aleph_succ, Ordinal.succ_zero]
+@[simp]
+theorem succ_aleph0 : succ ℵ₀ = ℵ₁ := by
+  rw [← aleph_zero, ← aleph_succ, Ordinal.succ_zero]
```
Comment (b-mehta) [inline Mathlib/SetTheory/Cardinal/Aleph.lean:452]:
```suggestion
@[simp] theorem succ_aleph0 : succ ℵ₀ = ℵ₁ := by rw [← aleph_zero, ← aleph_succ, Ordinal.succ_zero]
```

File: Mathlib/SetTheory/Cardinal/Aleph.lean [hunk 3]
```diff
@@ -417,6 +420,19 @@ theorem ord_aleph_isLimit (o : Ordinal) : (ℵ_ o).ord.IsLimit :=
...
   toType_noMax_of_succ_lt (ord_aleph_isLimit o).2
 
+@[simp]
+theorem range_aleph : range aleph = Set.Ici ℵ₀ := by
+  ext c
+  refine ⟨?_, fun hc ↦ ⟨preAleph.symm c - ω, ?_⟩⟩
+  · rintro ⟨o, rfl⟩
+    exact aleph0_le_aleph o
+  · rw [aleph_eq_preAleph, Ordinal.add_sub_cancel_of_le, preAleph.apply_symm_apply]
+    rwa [← aleph0_le_preAleph, preAleph.apply_symm_apply]
```
Comment (b-mehta) [inline Mathlib/SetTheory/Cardinal/Aleph.lean:430]:
```suggestion
  refine ⟨fun ⟨o, e⟩ => e ▸ aleph0_le_aleph _, fun hc ↦ ⟨preAleph.symm c - ω, ?_⟩⟩
  rw [aleph_eq_preAleph, Ordinal.add_sub_cancel_of_le, preAleph.apply_symm_apply]
  rwa [← aleph0_le_preAleph, preAleph.apply_symm_apply]
```

File: Mathlib/SetTheory/Cardinal/Aleph.lean (hunk 1 above)
Comment (vihdzp) [reply Mathlib/SetTheory/Cardinal/Aleph.lean:337]: I only added it for symmetry with `range_aleph`, but I don't really mind removing it.

Comment (b-mehta) [reply Mathlib/SetTheory/Cardinal/Aleph.lean:337]: Yeah let's get rid of it - `range_aleph` isn't immediately deduced from something else whereas this one is

**Review by j-loreaux — COMMENTED** [review]
Thanks!

bors merge

**Pull request closed without merging**